blocks/header/header.js

```diff
@@ -178,13 +194,32 @@ export default async function decorate(block) {
     brandLink.className = '';
     brandLink.closest('.button-container').className = '';
   }
-
+  const { body } = document;
   const navSections = nav.querySelector('.nav-sections');
   if (navSections) {
+    let prevNavSection;
     navSections.querySelectorAll(':scope .default-content-wrapper > ul > li').forEach((navSection) => {
       if (navSection.querySelector('ul')) navSection.classList.add('nav-drop');
       navSection.addEventListener('click', () => {
         if (isDesktop.matches) {
+          if (navSection.classList.contains('nav-drop')) {
+            if (navSection.classList.contains('active')) {
+              navSections.classList.remove('active');
+              navSection.classList.remove('active');
+              body.classList.remove('modal-open');
+            } else {
+              body.classList.add('modal-open');
+              prevNavSection?.classList.remove('active');
+              navSection.classList.add('active');
+              navSections.classList.add('active');
+              prevNavSection = navSection;
+            }
+          } else {
+            navSections.classList.remove('active');
+            navSection.classList.remove('active');
+            body.classList.remove('modal-open');
+            prevNavSection?.classList.remove('active');
+          }
```

Comment:
prefer to use 'aria-expanded'/ 'aria-hidden' (to show/hide content) true/false for accessibility purpose.

Reply:
done